zanata-war/src/main/java/org/zanata/webtrans/server/HibernateIntegrator.java

```diff
@@ -37,6 +38,8 @@ public void integrate(Configuration configuration,
                     updateListener);
             eventListenerRegistry.appendListeners(EventType.POST_INSERT,
                     updateListener);
+            eventListenerRegistry.appendListeners(EventType.POST_COMMIT_UPDATE,
+                    ServiceLocator.instance().getInstance(SlugEntityUpdatedListener.class));
```

Comment:
I guess this answers my previous question about how they play together.

But this gives me an idea, instead of having a single Application scoped bean and therefore worry about concurrency and stuff, can we have the listener itself be a POJO that calls a (preferably stateless but not necessarily) seam bean to do the event calling and all?